apps/nextjs/src/components/AppComponents/download/DownloadAllButton.tsx

```diff
@@ -0,0 +1,256 @@
+        >
+          {handleSendEmailIcon({ isSuccess, isLoading, isError })}
+          <div className="flex flex-col gap-6">
+            <span className="text-left font-bold">
+              Email me {ext === "docx" ? `doc` : `slides`}{" "}
+              {isSuccess && `- Email sent`}{" "}
+              {isError && `- There was an error sending the email!`}
+            </span>
+            <span className="text-left opacity-80">
+              Google account needed for this option
+            </span>
+          </div>
+        </button>
+      </div>
+    );
+  }
+  return (
+    <>
+      <button
+        className={`flex items-center justify-between gap-9 rounded-md border-2 border-black px-14 py-10 ${downloadAvailable ? `border-opacity-100` : ` border-opacity-40`}`}
+        onClick={() => onClick()}
+        disabled={!downloadAvailable}
+        data-testid={dataTestId}
+      >
+        <div className="flex items-center justify-start gap-7">
+          {handleIcon({
+            hasLink: !!link,
+            hasError: !!hasError,
+            downloadAvailable,
+            downloadLoading,
+            icon,
+          })}
+          <div className="flex flex-col items-start justify-start gap-6">
+            <span className="text-left font-bold">{title}</span>
+            {downloadLoading ? (
+              <span className="text-left text-[#287C34] opacity-80">
+                Generating {title} for export
+              </span>
+            ) : (
+              <span className="text-left opacity-80">{subTitle}</span>
+            )}
+          </div>
+        </div>
+        {!!link && (
+          <span className="flex items-center gap-5">
+            <span className="text-sm">Options</span>
+            <Icon icon="arrow-right" size="sm" />
+          </span>
+        )}
+      </button>
+      {hasError && <p className="mt-2 text-sm text-red-500">{errorMessage}</p>}
+    </>
+  );
+};
+
+function handleIcon({
+  hasLink,
+  hasError,
+  downloadAvailable,
+  downloadLoading,
+  icon,
+}: {
+  hasLink: boolean;
+  hasError: boolean;
+  downloadAvailable: boolean;
+  downloadLoading: boolean;
+  icon: "slides" | "lesson" | "quiz";
+}) {
+  if (hasLink) {
+    return null;
+  } else if (!downloadAvailable) {
+    return (
+      <Box className="w-20">
+        <Icon icon={"padlock"} className="mr-4" size="md" />
+      </Box>
+    );
+  } else if (downloadLoading) {
+    return (
+      <Box className="w-20">
+        <LoadingWheel />
+      </Box>
+    );
+  } else if (hasError) {
+    return (
+      <Box className="w-20">
+        <Icon icon={"chevron-down"} className="mr-4" size="sm" />
+      </Box>
+    );
+  } else if (downloadAvailable && !downloadLoading) {
+    return (
+      <Box className="mr-6 w-20">
+        {icon === "slides" && <SlidesIcon />}
+        {icon === "lesson" && <LessonIcon />}
+        {icon === "quiz" && <QuizIcon />}
+      </Box>
+    );
+  }
+  return null;
+}
+
+function handleSendEmailIcon({
```

Comment:
This looks kind of like a function component i.e. `<SendEmailIcon isError={} ... />`